File: 0714-best-time-to-buy-and-sell-stock-with-transaction-fee/0714-best-time-to-buy-and-sell-stock-with-transaction-fee.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices, int fee) {
        int s = 0, b = INT_MIN;
        for(auto x: prices){
            b = max(b,s-x);
            s = max(s,b+x-fee);
        }

        return s;
    }
};

File: 0714-best-time-to-buy-and-sell-stock-with-transaction-fee/0714-best-time-to-buy-and-sell-stock-with-transaction-fee.py
class Solution:
    def maxProfit(self, prices: List[int], fee: int) -> int:
        pos=-prices[0]
        profit=0
        for i in range(1,len(prices)):
            pos=max(pos,profit-prices[i])
            profit=max(profit,pos+prices[i]-fee)
        return profit